python/rascaline/power_spectrum.py

```diff
@@ -0,0 +1,201 @@
+from math import sqrt
+
+import numpy as np
+from equistore import Labels, TensorBlock, TensorMap
+
+
+try:
+    from equistore import slice_block
+except ImportError as err:
+    raise ImportError(
+        "equistore-operations are required to use the power spectrum."
+    ) from err
+
+
+def compute_power_spectrum(
+    spherical_expansion_1: TensorMap, spherical_expansion_2: TensorMap = None
+) -> TensorMap:
+    """Generate the rotationally invariant power spectrum from spherical expansions.
+
+    If two different sets of spherical expansions are provided, it is assumed that the
+    invariants are generated by taking quadratic combinations of those. Otherwise,
+    invariants are generated by combining coefficients of one spherical expansion.
```

Comment:
This is not extremely clear to me. How about putting the corresponding equation here? 

Also, the condition is somehow wrong: it is more about `spherical_expansion_2` being None than the two of them being the same, i.e. `compute_power_spectrum(spx, spx)` will still use the second code path.

Reply:
Another point to mention here is that in the single spx case there will be only half of the blocks, since (ns1, ns2) and (ns2, ns1) are symmetric.